packages/introspector/src/to-internal-struct.ts

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Session } from "neo4j-driver";
+import Node from "./classes/Node";
+import Property from "./classes/Property";
+import Relationship from "./classes/Relationship";
+import { Neo4jStruct, NodeMap, PropertyRecord, RelationshipMap } from "./types";
+import cleanTypeName from "./utils/clean-type-name";
+
+interface NodeTypePropertiesRecord extends PropertyRecord {
+    nodeType: string;
+    nodeLabels: string[];
+}
+interface RelationshipTypePropertiesRecord extends PropertyRecord {
+    relType: string;
+}
+
+type RelationshipRecord = {
+    from: string;
+    to: string;
+    relType: string;
+};
+
+export default async function toNeo4jStruct(sessionFactory: () => Session): Promise<Neo4jStruct> {
+    const nodes = await inferNodes(sessionFactory);
+    const relationships = await inferRelationships(sessionFactory);
+
+    return { nodes, relationships };
+}
+
+async function inferNodes(sessionFactory: () => Session): Promise<NodeMap> {
```

Comment:
I'm not going to highlight every one of these in the codebase, but worthwhile refactoring these remaining instances of "infer" to "instrospect" for consistency?

```suggestion
async function introspectNodes(sessionFactory: () => Session): Promise<NodeMap> {
```